packages/bundlers/default/src/DefaultBundler.js

```diff
@@ -151,6 +169,31 @@ export default new Bundler({
         bundleGraph.addAssetGraphToBundle(asset, bundle);
       }
     }
+
+    for (let [dependency, {bundleGroup, resolution}] of asyncDependencies) {
```

Comment:
Should this be done after optimization? Will shared bundles affect this?

Reply:
Probably. I don't think it should affect any of the deduplication or heuristics we currently use.

Reply:
Another thing is we could move the "internalizing" of dependencies in bundles into core, but it would make bundle group creation/deletion asymmetric. It also feels better to put this in the control of the bundler plugin, even though it feels like we're leaking a detail requiring the bundler to do this.